### ISSUE TITLE
Setting sync: Sign in dialogs not accessibility friendly

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode#120176]
From https://github.com/microsoft/vscode/issues/120012 by @neurrone

1. New vscode-insiders with fresh user data dir
2. Turn on settings sync, you get the Please Sign In quick pick dialog where you can check off which items you want to synchronize

This dialog is very unnacessible. Problems:
* Once the dialog is created the "Please sign in to synchronize" part is not being read out by screen reader. I recommend to use a placeholder since that would nicely be read.
* Down does not move focus to the Checkboxes
* It is hard for the user to figure out there are checkboxes below. Aria label of the input box should contain this information
* Checkbox list items do not have `aria-checked` set on them. For example on how to do this, checkout [breakpoints](https://github.com/microsoft/vscode/blob/main/src/vs/workbench/contrib/debug/browser/breakpointsView.ts#L848) 

fyi @chrmarti @TylerLeonhardt 

![Screenshot 2021-03-30 at 15 02 47](https://user-images.githubusercontent.com/1926584/112993246-45963e80-9169-11eb-82f3-fbbb2b2969b1.png)

**Comment by isidorn** [comment]
Comments from @neurrone

I initially thought the text box was asking for some sort of username until I realized that its actually meant to filter the list of 5 categories of settings you can sync. That was very confusing because at least with a screen reader, its not obvious at all. Its also right next to the sign in button in tab order, which reinforced the impression that you're supposed to enter some sort of username

**Comment by isidorn** [comment]
Please note that the Stable vs Insiders dialogs suffers from the same problems. It should be fixed in the same way

![Screenshot 2021-03-30 at 15 11 12](https://user-images.githubusercontent.com/1926584/112994106-3368d000-916a-11eb-9571-ded231119df8.png)

**Comment by sandy081** [comment]
@chrmarti / @TylerLeonhardt I am just using Quick input API here, May I know if I have to pass any thing to the API to make this accessible as suggested by @isidorn ?

**Comment by chrmarti** [comment]
> * Once the dialog is created the "Please sign in to synchronize" part is not being read out by screen reader. I recommend to use a placeholder since that would nicely be read.

@sandy081 Could you use a placeholder here? The combination of title and prompt looks very verbose.

> * Down does not move focus to the Checkboxes

@TylerLeonhardt This might be a bug, I thought I had this working.

> * It is hard for the user to figure out there are checkboxes below. Aria label of the input box should contain this information
> * Checkbox list items do not have `aria-checked` set on them. For example on how to do this, checkout [breakpoints](https://github.com/microsoft/vscode/blob/main/src/vs/workbench/contrib/debug/browser/breakpointsView.ts#L848)

These can be improved.

> I initially thought the text box was asking for some sort of username until I realized that its actually meant to filter the list of 5 categories of settings you can sync. That was very confusing because at least with a screen reader, its not obvious at all. Its also right next to the sign in button in tab order, which reinforced the impression that you're supposed to enter some sort of username

@sandy081 Maybe there should be an additional step for "Sign in & Turn On"?

**Comment by Neurrone** [comment]
> Maybe there should be an additional step for "Sign in & Turn On"?

Yes, that's what I was thinking of as well, though it does create one additional prompt. But this is only soomething that the user would likely see once, upon enabling settings sync.

**Comment by sandy081** [comment]
> Could you use a placeholder here? The combination of title and prompt looks very verbose.

Actually I do not show input/text box in that picker and I see following

![image](https://user-images.githubusercontent.com/10746682/113147007-a472bb00-9230-11eb-9e80-51f8724ec1b7.png)

Not sure how @isidorn is seeing the one with input box.

**Comment by isidorn** [comment]
@sandy081 yes I see the input box. If somebody wants to debug on my machine feel free to ping me.

**Comment by sandy081** [comment]
Lets debug together then :)

**Comment by sandy081** [comment]
Investigated with @isidorn and it seems the input box is shown when following setting is set `"editor.accessibilitySupport": "on"`.

@chrmarti @TylerLeonhardt - To summarize there are following issues with quick pick

1. Input box is shown when following setting is set `"editor.accessibilitySupport": "on"` - even if it is asked not to be shown.
2. Quick pick is not reading the description

**Comment by TylerLeonhardt** [comment]
verification steps:

1. apply this setting: `"editor.accessibilitySupport": "on"`
2. turn on a screen reader
3. go through the "turn on settings sync" flow

You shouldn't see the input box for the quick picks shown above, and the screenreader should announce the description (the text under the title).

**Comment by isidorn** [comment]
This is better now -> verified
However I created one follow up item based on feedback from @jvesouza

**Comment by Neurrone** [comment]
Is there a way for me to get a portable copy of a build containing these changes so I can go through the first time settings sync setup experience again without removing my current sync settings?

**Comment by isidorn** [comment]
@Neurrone I suggest to just start vscode insiders from the command line pointing to an empty user data dir. Like this `vscode-insiders --user-data-dir PATH_TO_SOME_EMPTY_FOLDER`

**Comment by Neurrone** [comment]
This doesn't seem to be fixed.

After turning it on and pressing OK after the dialog box saying that this is an empty feature:

1. I find that the arrow keys do let me navigate the list of categories of settings to sync, but the selected state is not read. Only if i press tab does it get read, as tab seems to also be able to bring focus to each individual list item. That seems like a bug because I've never seen this keyboard control pattern anywhere else.
2. I also see on that same dialog the button that says "sign in to turn on", but the help text that explains why the signing in is required is also not read. I'm wondering if its just going to be easier to split these 2 questions (selection of categories) + the sign in prompt into separate dialogs?

**Comment by isidorn** [comment]
@Neurrone thanks for letting us know
@TylerLeonhardt let me know if you need help here

**Comment by TylerLeonhardt** [comment]
This one can be closed out now. We do implement `isChecked` correctly now and the correct state is read. Verification:

1. Screen reader on
2. A11y mode on in VS Code
1. `>Settings Sync: Configure...`
2. toggle the checkboxes using the keyboard

Ensure the screen reader says they're state correctly (checked or unchecked)
